weldx/tests/test_time.py

```diff
@@ -375,6 +374,229 @@ def test_add_timedelta(
         assert np.all(res.as_timedelta() == exp.as_timedelta())
         assert np.all(res == exp)
 
+    # test_add_datetime ----------------------------------------------------------------
+
+    @staticmethod
+    @pytest.mark.parametrize("other_on_rhs", [True, False])
+    @pytest.mark.parametrize("time_class_is_array", [False, True])
+    @pytest.mark.parametrize("other_is_array", [False, True])
+    @pytest.mark.parametrize(
+        "other_type",
+        [
+            str,
+            Time,
+            Q_,
+            TDI,
+            Timedelta,
+            np.timedelta64,
+        ],
+    )
+    def test_add_datetime(
+        other_type,
+        other_on_rhs: bool,
+        time_class_is_array: bool,
+        other_is_array: bool,
+    ):
+        """Test the `__add__` method if the `Time` class represents a datetime.
+
+        Parameters
+        ----------
+        other_type :
+            The type of the other object
+        other_on_rhs :
+            If `True`, the other type is on the rhs of the + sign and on the lhs
+            otherwise
+        time_class_is_array :
+            If `True`, the `Time` instance contains 3 time values and 1 otherwise
+        other_is_array :
+            If `True`, the other time object contains 3 time values and 1 otherwise
+
+        """
+        # skip array cases where the type does not support arrays
+        if other_type in [Timedelta, Timestamp] and other_is_array:
+            pytest.skip()
```

Comment:
the `pytest.skip()` command immediately jumps out of the function here, right?
In that case I prefer to simply `return` and reserve `skip` for cases where the whole test case did not start for a cleaner pytest output

Reply:
test it please and update accordingly